data/src/test/java/io/gnosis/data/repositories/TransactionRepositoryTest.kt

```diff
@@ -0,0 +1,391 @@
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.*
+import org.junit.Test
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asDecimalString
+import pm.gnosis.utils.asEthereumAddress
+import java.math.BigInteger
+
+class TransactionRepositoryTest {
+
+    private val transactionServiceApi = mockk<TransactionServiceApi>()
+
+    private val transactionRepository = TransactionRepository(transactionServiceApi)
+    private val defaultSafeAddress = "0x1C8b9B78e3085866521FE206fa4c1a67F49f153A".asEthereumAddress()!!
+    private val defaultFromAddress = "0x7cd310A8AeBf268bF78ea16C601F201ca81e84Cc".asEthereumAddress()!!
+    private val defaultToAddress = "0x2134Bb3DE97813678daC21575E7A77a95079FC51".asEthereumAddress()!!
+    private val defaultValue = BigInteger("230000000000000000")
+
+    @Test
+    fun `getTransactions (api failure) should throw`() = runBlockingTest {
+        val throwable = Throwable()
+        coEvery { transactionServiceApi.loadTransactions(any()) } throws throwable
+
+        val actual = runCatching { transactionRepository.getTransactions(defaultSafeAddress) }
+
+        with(actual) {
+            assert(isFailure)
+            assertEquals(throwable, exceptionOrNull())
+        }
+        coVerify(exactly = 1) { transactionServiceApi.loadTransactions(defaultSafeAddress.asEthereumAddressChecksumString()) }
+    }
+
+    @Test
+    fun `getTransactions (module transaction) should return custom`() = runBlockingTest {
+        val transactionDto = buildModuleTransactionDto(module = Solidity.Address(BigInteger.ONE))
+        val pagedResult = listOf(transactionDto)
+        coEvery { transactionServiceApi.loadTransactions(any()) } returns Page(1, null, null, pagedResult)
+
+        val actual = transactionRepository.getTransactions(defaultSafeAddress)
+
+        with(actual.results[0] as Transaction.Custom) {
+            assertEquals(transactionDto.module, address)
+            assertEquals(transactionDto.data?.dataSizeBytes() ?: 0L, dataSize)
+            assertEquals(transactionDto.created, date)
+            assertEquals(transactionDto.value, value)
+        }
+    }
+
+    @Test
+    fun `getTransactions (UnknownTransaction with data) should return custom`() = runBlockingTest {
+        val transactionDto = UnknownTransactionDto
+        val pagedResult = listOf(transactionDto)
+        coEvery { transactionServiceApi.loadTransactions(any()) } returns Page(1, null, null, pagedResult)
+
+        val actual = transactionRepository.getTransactions(defaultSafeAddress)
+
+        assertEquals(1, actual.results.size)
+        with(actual.results[0] as Transaction.Custom) {
+            assertEquals(transactionDto.to, address) //TODO: Only to available in UnknownTransaction, but is it useful?
```

Comment:
There should be enough fields to be able to render a `Custom` transaction.

Reply:
I removed the comment